build.gradle

```diff
@@ -44,6 +44,12 @@ tasks.withType(JavaCompile).configureEach {
     options.release.set(javaTargetVersion.majorVersion.toInteger())
 }
 
+tasks.withType(Test).configureEach {
+    doLast {
+        Thread.sleep(2000) // https://github.com/gradle/gradle/issues/16603
+    }
+}
```

Comment:
I'm not 100% sure if it is the same issue, but given I experienced something similar, happening only on Windows, not reproducible locally, taking place only after all tests complete successfully - I'm proposing a workaround that was inspired by one of PRs referenced in the linked issue. It's been [working for me](https://github.com/usefulness/project-starter/commit/ff75d01fffe2fe7c2628b1121abd4145335a06e3) for the last 2 years, and it seems like I already got 2 successful builds :)

Seeing this issue [had been](https://github.com/koral--/jacoco-gradle-testkit-plugin/pull/20) happening already in the past, my understanding is the workaround wasn't complete, and one of my changes ([enabling build-cache](https://github.com/koral--/jacoco-gradle-testkit-plugin/pull/26/files#diff-3d103fc7c312a3e136f88e81cef592424b8af2464c468116545c4d22d6edcf19R18)?) increased its probability, or I was just unlucky since the same set of changes had already passed the CI, before the rebase 🤷 

Anyway, this is my proposal, the only alternative I can think of would be to disable the build cache

Reply:
Also, this assumes the `--project-cache-dir` was added only to work around the issue 👀 If the CI actually relies on the `../cache` then please let me know!